### DISCUSSION TITLE
input validation & its own fill when dragged down

### DISCUSSION THREAD

**Discussion by phoique** [iddan/react-spreadsheet#240] [Q&A] [answered]
Hi, I love using this package. I have 2 questions, my first question is that I want to enter only numbers and not onChange when entering letters. Is there an easy way to do this difficult step because the matrix is returned in onchange?

My other question is that when 10:20 is entered as 10:30 and dragged down, I want to fill it as 10:40, 10:50, 11:00. Is it possible for me to do this?

**Comment by iddan** [comment, marked as answer]
Hey @phoique. You can use a custom `<DataEditor />` please see the storybook for reference. About the other one, what do you mean by dragged down?

**Reply by phoique** [reply]
Ahh. I'm so forgetful and careless these days. I solved the input problem with Data Editor. I am sharing an image to be more descriptive of my other question.

[https://ibb.co/MD2qSWM](url)

If we consider it as 10:20 and 10:25 hours in the image, when I drag it down from 10:20, I want it to continue as 10:30, 10:35 with 2 empty spaces under it.

**Reply by iddan** [reply]
I see. I think custom paste logic is not supported yet but maybe you can achieve this in `onChange` by comparing the previous and new states?

**Reply by phoique** [reply]
I'm sorry you didn't support it directly. I discovered a solution. onSelect is a function triggered on multiple selection. It returns the row and column value of the selected one. This is super stuff. The first 2 values returned in onSelect are the columns that will determine my range. I can fill other columns according to that range.

I am very grateful for your help. A special thank you for your very quick response.
